runway/module/__init__.py

```diff
@@ -60,67 +24,169 @@ def run_module_command(cmd_list, env_vars):
-        raise NotImplementedError('You must implement the plan() method '
-                                  'yourself!')
+        raise NotImplementedError('You must implement the plan() method yourself!')
 
     def deploy(self):
         """Implement dummy method (set in consuming classes)."""
-        raise NotImplementedError('You must implement the deploy() method '
-                                  'yourself!')
+        raise NotImplementedError('You must implement the deploy() method yourself!')
 
     def destroy(self):
         """Implement dummy method (set in consuming classes)."""
-        raise NotImplementedError('You must implement the destroy() method '
-                                  'yourself!')
+        raise NotImplementedError('You must implement the destroy() method yourself!')
+
+
+class RunwayModuleFolder(object):
+    """Functions to manage filesystem access in a module folder."""
+
+    def __init__(self, path):
+        """Initialize base class."""
+        self._path = path
+
+    def fullpath(self, name):
+        """Return the absolute path to the given file."""
+        return os.path.join(self._path, name)
+
+    def isfile(self, name):
+        """Determine if the given file exist relative to the module."""
+        return name and os.path.isfile(self.fullpath(name))
+
+    def isdir(self, name):
+        """Determine if the given folder exist relative to the module."""
+        return name and os.path.isdir(self.fullpath(name))
+
+    def locate_file(self, names):
+        """Given a list of files, find one that exists (if any) in the root folder."""
+        for name in names:
+            if self.isfile(name):
+                return name
+        return None
+
+    def locate_env_file(self, names):
+        """Given a list of files, find one that exists (if any) in the root or `env` folders."""
+        # first try in the root of the module folder
+        location = self.locate_file(names)
+        if not location:
+            # next try in the 'env' folder
+            env_names = [os.path.join('env', name) for name in names]
+            location = self.locate_file(env_names)
+        return location
+
+    def load_json_file(self, name):
+        """Load the contents of the JSON file into a dict."""
+        with open(self.fullpath(name), 'r') as stream:
+            return json.load(stream)
+
+    def load_yaml_file(self, name):
+        """Load the contents of the YAML file into a dict."""
+        with open(self.fullpath(name), 'r') as stream:
+            # load() returns None on an existing but empty file, so provide a default
+            return yaml.load(stream) or {}
+
+
+class NpmHelper(object):
+    """Functions to wrap around npm for use by the various modules."""
+
+    def __init__(self, module_name, project_options, env_vars, folder):
+        """Create an instance."""
+        self._module_name = module_name
+        self._project_options = project_options
```

Comment:
This would become something like `module_custom_options` based on my deprecation suggestion above.